js/plugins/dotprompt/src/registry.ts

```diff
@@ -80,15 +80,23 @@ export async function loadPromptFolder(
         promptsPath,
         {
           withFileTypes: true,
-          recursive: false,
+          recursive: true,
         },
         (err, dirEnts) => {
           if (err) {
             reject(err);
           } else {
             dirEnts.forEach(async (dirEnt) => {
               if (dirEnt.isFile() && dirEnt.name.endsWith('.prompt')) {
-                loadPrompt(dirEnt.path, dirEnt.name);
+                // If this prompt is in a subdirectory, we need to include that
+                // in the namespace to prevent naming conflicts.
+                let subDir = '';
+                if (promptsPath !== dirEnt.path) {
+                  subDir = dirEnt.path
+                    .replace(`${promptsPath}/`, '')
+                    .replace(/\//g, '-');
+                }
```

Comment:
would this make sense to be inside `loadPrompt`? I mean, you're passing it as `subDir`, but it's not more of a prefix... alternatively maybe just rename `subDir` to `prefix`?

Reply:
We would need to pass in `promptsPath` for the extra logic, so it's a sixer to me. Good call on the naming, updated to `prefix`.